### PR TITLE
Support Spring Boot 2.0

### PR DESCRIPTION
**Pull request by ddcruver**
This should address #25.

### CODE REVIEW

File: spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/AnnotationConfigNonEmbeddedWebApplicationContext.java [hunk 1]
```diff
@@ -72,19 +82,34 @@ public AnnotationConfigNonEmbeddedWebApplicationContext() {
...
 	 * context.
-	 * @param annotatedClasses one or more annotated classes, e.g. {@link Configuration
-	 * &#64;Configuration} classes
+	 * @param annotatedClasses one or more annotated classes, e.g. {@code @Configuration}
+	 * classes
 	 */
 	public AnnotationConfigNonEmbeddedWebApplicationContext(
-			Class<?>... annotatedClasses) {
+			Class<?>... annotatedClasses)
+	{
```
Comment (dsyer) [inline spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/AnnotationConfigNonEmbeddedWebApplicationContext.java]: Not a big deal really, but we generally don't put opening braces on a new line in Spring projects.

File: spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/SpringBootContextLoaderListener.java [hunk 2]
```diff
@@ -28,24 +30,30 @@
...
  * of all the initialization extras in Spring Boot even if they don't use an embedded
  * container.
- * 
+ *
+ * @author Daniel Cruver
  * @author Dave Syer
  */
 public class SpringBootContextLoaderListener extends ContextLoaderListener {
 
+	private static final Logger log = LoggerFactory.getLogger(SpringBootContextLoaderListener.class);
```
Comment (dsyer) [inline spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/SpringBootContextLoaderListener.java]: Please use commons-logging for Spring libraries.

File: spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/AnnotationConfigNonEmbeddedWebApplicationContext.java (hunk 1 above)
Comment (ddcruver) [reply spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/AnnotationConfigNonEmbeddedWebApplicationContext.java]: Will fix, yeah that was my normal projects code style sneaking in.

File: spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/SpringBootContextLoaderListener.java (hunk 2 above)
Comment (ddcruver) [reply spring-boot-legacy/src/main/java/org/springframework/boot/legacy/context/web/SpringBootContextLoaderListener.java]: Will fix.

**Comment by ddcruver** [comment]
Actually just the Java 7 Runtime is deprecated.

spring-boot-sample-gae  - Java 7 Runtime
spring-boot-sample-gae-2.5 - Java 8 Runtime with Servlet 2.5 Web Descriptor
spring-boot-sample-gae-3.0 - Java 8 Runtime with Servlet 3.0 Web Descriptor
spring-boot-sample-gae-3.1 - Java 8 Runtime with Servlet 3.1 Web Descriptor

Maybe we should delete the spring-boot-sample-gae and spring-boot-sample-gae-3.0.  

The first being the one that is already deprecated and actually doesn't work, because of Spring Boot 2.0 requiring Java 8.  If someone wants Spring Boot in a Java 7 environment they can use an older release based on your new legacy branch you created and be stuck on the previous version of Spring Boot.

The later, spring-boot-sample-gae-3.0 shows you can use a 3.0 web.xml and it will run a Servlet 3.0. Since Google AppEngine officially supports Servlet 3.1, makes no mention of Servlet 3.0, maybe anyone developing Spring Boot apps to run on Google AppEngine should most likely use the 3.1 web.xml anyways.

**Comment by dsyer** [comment]
> Maybe we should delete the spring-boot-sample-gae and spring-boot-sample-gae-3.0.

Makes sense to me. I see very little value in either.

**Comment by ddcruver** [comment]
@dsyer I think those changes should make this pull request good.

**Comment by dsyer** [comment]
Merged, thanks. I couldn't get the gae-2.5 sample to work on AppEngine, but the 3.1 sample seems OK. Any hints on what to do? (The README is copy pasted from another project I think.)

**Comment by ddcruver** [comment]
I used appengine:devserver to test everything. I will double check but I didn't believe I had any problems. I was also running with a Java 8 JDK as well.

In the other project I deleted I was getting an error with java.xxx.streams because it was not a part of the whitelist and was trying to get it to work in the Java 7 environment. Is it possibly this project is still running in the Java 7 Runtime when deployed to Google AppEngine?

**Pull request closed without merging**